### DISCUSSION TITLE
How to add Express server around RR7?

### DISCUSSION THREAD

**Discussion by DennisKraaijeveld** [remix-run/react-router#12661] [General]
Is there any documentation on how to wrap RR7 icw Vite, with a custom server like we could do with Remix?

```
"dev": "react-router dev",
```

entry file:

```
// MSW / Mocking
if (process.env.MOCKS === 'true') {
	await import('./tests/mocks/index.ts')
}

// Prod Server
if (process.env.NODE_ENV === 'production') {
	await import('./server-build/index.js')
} else {
// Dev Server
	await import('./server/index.ts')
}
```

Should I somehow set this up within the vite config now?

**Comment by gustavopch** [comment]
You can follow this: https://github.com/remix-run/react-router/tree/08e4f2fd399543cab776f4be8a29181093a3702c/playground/framework-express

**Reply by Tom-DK** [reply]
Any typescript example? 🥹
